tests/cache_test.py

```diff
@@ -113,11 +100,57 @@ def do_something():
             cache.refresh()  # force refresh
             do_something()  # won't fail anymore
 
+    def test_main_with_explicit_refresh_of_group(self):
+        group = SSMParameterGroup()  # will not expire
+        param_1 = group.parameter("my_param_1")
+        param_2 = group.parameter("my_param_2")
+
+        class InvalidCredentials(Exception):
+            pass
+
+        def do_something():
+            my_value = param_1.value
+            if my_value == self.PARAM_VALUE:
+                raise InvalidCredentials()
+
+        try:
+            do_something()
+        except InvalidCredentials:
+            # manually update value
+            NEW_VALUE = "new_value"
+            self._create_params(["my_param_1", "my_param_2"], NEW_VALUE)
+            group.refresh()  # force refresh
+            do_something()  # won't fail anymore
+            self.assertEqual(param_2.value, NEW_VALUE)
+
+    def test_main_with_explicit_refresh_of_group_param(self):
+        group = SSMParameterGroup()  # will not expire
+        param_1 = group.parameter("my_param_1")
+        param_2 = group.parameter("my_param_2")
+
+        class InvalidCredentials(Exception):
+            pass
+
+        def do_something():
+            my_value = param_1.value
+            if my_value == self.PARAM_VALUE:
+                raise InvalidCredentials()
+
+        try:
+            do_something()
+        except InvalidCredentials:
+            # manually update value
+            NEW_VALUE = "new_value"
+            self._create_params(["my_param_1", "my_param_2"], NEW_VALUE)
+            param_1.refresh()  # force refresh
```

Comment:
Based on what we decide on the refresh logic above (here I'd expect to refresh only param_1 and not the entire group), it'd be great to verify that only 1 API call is performed here as well.